src/ReactiveMarbles.PropertyChanged/GetMemberFuncCache.cs

```diff
@@ -12,35 +12,33 @@ namespace ReactiveMarbles.PropertyChanged
     internal static class GetMemberFuncCache<TFrom, TReturn>
     {
 #if !UIKIT
-        private static readonly
-            ConcurrentDictionary<(Type FromType, string MemberName), Func<TFrom, TReturn>> Cache
-                = new ConcurrentDictionary<(Type, string), Func<TFrom, TReturn>>();
+        private static readonly ConcurrentDictionary<MemberInfo, Func<TFrom, TReturn>> Cache = new(new MemberFuncCacheKeyComparer());
 #endif
 
         public static Func<TFrom, TReturn> GetCache(MemberInfo memberInfo)
         {
 #if UIKIT
-                switch (memberInfo)
-                {
-                    case PropertyInfo propertyInfo:
-                        return input => (TReturn)propertyInfo.GetValue(input);
-                    case FieldInfo fieldInfo:
-                        return input => (TReturn)fieldInfo.GetValue(input);
-                    default:
-                        throw new ArgumentException($"Cannot handle member {memberInfo.Name}", nameof(memberInfo));
-                }
+            switch (memberInfo)
+            {
+                case PropertyInfo propertyInfo:
+                    return input => (TReturn)propertyInfo.GetValue(input);
+                case FieldInfo fieldInfo:
+                    return input => (TReturn)fieldInfo.GetValue(input);
+                default:
+                    throw new ArgumentException($"Cannot handle member {memberInfo.Name}", nameof(memberInfo));
+            }
 #else
-            return Cache.GetOrAdd((memberInfo.DeclaringType, memberInfo.Name), _ =>
+            return Cache.GetOrAdd(memberInfo, static mi =>
             {
                 var instance = Expression.Parameter(typeof(TFrom), "instance");
 
-                var castInstance = Expression.Convert(instance, memberInfo.DeclaringType);
+                var castInstance = Expression.Convert(instance, mi.DeclaringType);

```

Comment:
usually prefer not to use abbreviated names for this type of thing.

Reply:
Alright, I'll change it. I don't prefer abbreviated names either, but I wanted to differentiate it from the method parameter. But not a big deal since the static keyword will enforce it anyways.

Reply:
Yeah potentially you could a prefix or something?

Reply:
I think using the same name is fine (methodInfo). _ prefix is for private variables, and I know you hate the @ prefix.